app/src/main/java/com/android/ubclaunchpad/driver/session/SessionInfoActivity.java

```diff
@@ -215,14 +180,148 @@ public void onCancelled(DatabaseError databaseError) {
+
+            }
+        };
+
+
+
+        driverPassengersListener = new ChildEventListener() {
+
+            @Override
+            public void onChildAdded(final DataSnapshot sessionSnapshot, String s) {
+
+                DatabaseReference usersReference = FirebaseUtils.getDatabase().child(StringUtils.FirebaseUserEndpoint);
+                usersReference.addListenerForSingleValueEvent(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(DataSnapshot usersSnapshot) {
+
+                        // If algorithm finished
+                        if (sessionSnapshot.getKey().equals(StringUtils.FirebaseSessionDriverPassengers)) {
+
+                            User currentUser = usersSnapshot.child(UID).getValue(User.class);
+                            DataSnapshot driverPassengersSnapshot = sessionSnapshot.child(StringUtils.FirebaseSessionDriverPassengers);
+                            Bundle data = new Bundle();
+                            Message message = new Message();
+                            message.setData(data);
+                            data.putString(StringUtils.FirebaseSessionName, sessionName);
+                            if (currentUser.getIsDriver()) {
+
+                                // No need to search for anything, just start DriverPassengersActivity
+                                data.putString("driverUid", UID);
+                                runOnUiThread(startActivityTask(data));
+                            } else {
+                                // We need to find to which driver this passenger belongs to
+                                for (DataSnapshot driver : driverPassengersSnapshot.getChildren()) {
+                                    for (DataSnapshot passenger : driver.getChildren()) {
+                                        if (passenger.getValue().equals(UID)) {
+
+                                            data.putString("driverUid", UID);
+                                            runOnUiThread(startActivityTask(data));
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    @Override
+                    public void onCancelled(DatabaseError databaseError) {
+
+                    }
+                });
+            }
+
+            @Override
+            public void onChildChanged(DataSnapshot dataSnapshot, String s) {
+
+            }
+
+            @Override
+            public void onChildRemoved(DataSnapshot dataSnapshot) {
+
+            }
+
+            @Override
+            public void onChildMoved(DataSnapshot dataSnapshot, String s) {
+
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+
+            }
+        };
+    }
+
+    private Runnable startActivityTask(final Bundle data) {
```

Comment:
can you call this something like startDriverPassengersActivityTask instead since it always starts that specific activity?